compiler-core/src/type_/prelude.rs

```diff
@@ -84,13 +84,23 @@ pub fn float() -> Arc<Type> {
 }
 
 pub fn bool() -> Arc<Type> {
+    bool_with_variant(None)
+}
+
+pub fn bool_with_variant(variant: Option<bool>) -> Arc<Type> {
```

Comment:
It says with variant so always take the variant please 🙏 
```suggestion
pub fn bool_with_variant(variant: bool) -> Arc<Type> {
```